spec/07-variables.md

```diff
@@ -50,6 +50,14 @@ A constant defined outside of a class or interface is a superglobal
 ([§§](#general)). A constant has static storage duration ([§§](04-basic-concepts.md#storage-duration)) 
 and is a non-modifiable lvalue.
 
+**Undefined Constants**
+
+PHP does not define undefined constants implicitly -- forward usages of constants are also classified as undefined constants here. A distinction between class/interface constants and top level constants is made.
+
+For top level constants: the name of the undefined constant (as string) is used as substitution value. Moreover, an  `E_NOTICE` is emitted stating that the corresponding constant was undefined.
+
+For class/interface constants: an `E_FATAL_ERROR` is produced stating that the corresponding constant was undefined.
```

Comment:
The spec does not use specific error constants, instead "fatal error" is used, as defined in 03-terms-and-definitions.md. Same for notices. While we do mention E_* constants as defined, E_FATAL_ERROR is not one of them, and there's no need to actually mention them and confuse constants with actual notices. What is produced is a fatal error, not an E_* constant.